### PR TITLE
Add feature flag-based lemonade opt-in

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

Achieving (transitional) world peace by the means of an opt-in-or-out switch. No new columns, pure person properties (`opted_into_lemonade: boolean`).

### CODE REVIEW

**Review by mariusandra — APPROVED** [review]
LGTM. Feel free to merge whenever it makes sense...

**Comment by Twixes** [comment]
Polished a few more things and added a nice switch component based on @clarkus's design.

**Review by mariusandra — APPROVED** [review]

**Comment by Twixes** [comment]
With a spinner.

File: frontend/src/layout/lemonade/RedesignOptIn.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                {featureFlags[FEATURE_FLAGS.LEMONADE]
+                    ? "You're using the new design"
+                    : "Try PostHog's navigation redesign!"}
+            </label>
+            <LemonSwitch
+                id="redesign-opt-in"
+                onChange={(checked) => {
+                    setIsSwitchLoading(true)
+                    setIsSwitchChecked(checked)
+                    posthog.people.set('opted_into_lemonade', checked, () =>
```
Comment (paolodamico) [inline frontend/src/layout/lemonade/RedesignOptIn.tsx:29]: Late posthumous review, we should've added an event too to report when users opt-in/out as we don't have properties over time we won't know for instance if a user enabled it, got scared and went back. Autocapture will also not be perfect as we won't know for sure the value of the switch when clicked.

Comment (Twixes) [reply frontend/src/layout/lemonade/RedesignOptIn.tsx:29]: Don't we already do this with the `posthog.capture(checked ? `opted into navigation redesign` : 'opted out of navigation redesign')` line below? I've been using that data

**Pull request merged**